### PR TITLE
Upgrade to netcoreapp22

### PR DESCRIPTION
**Pull request by ryanbrandenburg**

### CODE REVIEW

**Comment by halter73** [comment]
@ryanbrandenburg When you merge this can you also update the Dockerfiles, scripts and readmes? There's only 5 more places that I know of. You can see them here: #2542

File: src/Kestrel.Core/Kestrel.Core.csproj [hunk 1]
```diff
@@ -1,10 +1,10 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
+<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <AssemblyName>Microsoft.AspNetCore.Server.Kestrel.Core</AssemblyName>
     <RootNamespace>Microsoft.AspNetCore.Server.Kestrel.Core</RootNamespace>
     <Description>Core components of ASP.NET Core Kestrel cross-platform web server.</Description>
-    <TargetFrameworks>netstandard2.0;netcoreapp2.1</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;netcoreapp2.2</TargetFrameworks>
```
Comment (Tratcher) [inline src/Kestrel.Core/Kestrel.Core.csproj]: Also https://github.com/aspnet/KestrelHttpServer/blob/8b1fbad10e12e7ddab98ae7a17891eca6e6bcfd9/src/Kestrel.Core/Internal/HttpsConnectionAdapter.cs#L129

File: Directory.Build.targets [hunk 2]
```diff
@@ -1,6 +1,8 @@
-﻿<Project>
+<Project>
   <PropertyGroup>
     <RuntimeFrameworkVersion Condition=" '$(TargetFramework)' == 'netcoreapp2.0' ">$(MicrosoftNETCoreApp20PackageVersion)</RuntimeFrameworkVersion>
```
Comment (Tratcher) [inline Directory.Build.targets]: Cleanup

File: src/Kestrel.Core/Kestrel.Core.csproj (hunk 1 above)
Comment (Tratcher) [reply src/Kestrel.Core/Kestrel.Core.csproj]: Actually, this one likely needs to stay at 2.1. It was only added for new API. There's no reason to move it higher right now.

File: src/Kestrel.Https/Kestrel.Https.csproj [hunk 3]
```diff
@@ -1,10 +1,10 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
+<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <AssemblyName>Microsoft.AspNetCore.Server.Kestrel.Https</AssemblyName>
     <RootNamespace>Microsoft.AspNetCore.Server.Kestrel.Https</RootNamespace>
     <Description>HTTPS support for the ASP.NET Core Kestrel cross-platform web server.</Description>
-    <TargetFrameworks>netstandard2.0;netcoreapp2.1</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;netcoreapp2.2</TargetFrameworks>
```
Comment (Tratcher) [inline src/Kestrel.Https/Kestrel.Https.csproj]: Leave at 2.1

File: src/Kestrel.Transport.Sockets/Kestrel.Transport.Sockets.csproj [hunk 4]
```diff
@@ -1,10 +1,10 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
+<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <AssemblyName>Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets</AssemblyName>
     <RootNamespace>Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets</RootNamespace>
     <Description>Managed socket transport for the ASP.NET Core Kestrel cross-platform web server.</Description>
-    <TargetFrameworks>netstandard2.0;netcoreapp2.1</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;netcoreapp2.2</TargetFrameworks>
```
Comment (Tratcher) [inline src/Kestrel.Transport.Sockets/Kestrel.Transport.Sockets.csproj]: Leave at 2.1

File: build/repo.props [hunk 5]
```diff
@@ -10,6 +10,6 @@
 
   <ItemGroup>
     <DotNetCoreRuntime Include="$(MicrosoftNETCoreApp20PackageVersion)" />
```
Comment (Tratcher) [inline build/repo.props]: Cleanup

File: build/dependencies.props [hunk 6]
```diff
@@ -26,6 +26,7 @@
     <MicrosoftExtensionsOptionsPackageVersion>2.1.0-preview1-27965</MicrosoftExtensionsOptionsPackageVersion>
     <MicrosoftNETCoreApp20PackageVersion>2.0.0</MicrosoftNETCoreApp20PackageVersion>
     <MicrosoftNETCoreApp21PackageVersion>2.1.0-preview1-26102-01</MicrosoftNETCoreApp21PackageVersion>
```
Comment (Tratcher) [inline build/dependencies.props]: cleanup

File: src/Kestrel.Core/Adapter/Internal/RawStream.cs [hunk 7]
```diff
@@ -69,7 +69,7 @@ public override Task<int> ReadAsync(byte[] buffer, int offset, int count, Cancel
             return ReadAsyncInternal(new Memory<byte>(buffer, offset, count)).AsTask();
         }
 
-#if NETCOREAPP2_1
+#if NETCOREAPP2_2
```
Comment (Tratcher) [inline src/Kestrel.Core/Adapter/Internal/RawStream.cs]: Product assemblies should only move forward to 2.2 as needed.

Comment (ryanbrandenburg) [reply src/Kestrel.Core/Adapter/Internal/RawStream.cs]: Correct, nice catch.

**Comment by ryanbrandenburg** [comment]
:up::date: @halter73 and @Tratcher if you wanted to double check me since I had to mess with a lot more here than on most repos.

**Pull request merged**